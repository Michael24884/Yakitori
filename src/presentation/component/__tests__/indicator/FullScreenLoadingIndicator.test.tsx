/**
 * @format
 */
import React from 'react';
import {Modal} from 'react-native';

import {
  FullScreenLoadingIndicator,
  FullScreenLoadingIndicatorProps,
} from '@components';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('test visible', () => {
  const props: FullScreenLoadingIndicatorProps = {
    visible: true,
  };
  const instance = renderer.create(<FullScreenLoadingIndicator {...props} />);
  expect(instance.root.findByType(Modal).props?.visible).toBeTruthy();
});

it('render correctly', () => {
  const props: FullScreenLoadingIndicatorProps = {
    visible: true,
  };
  const instance = renderer.create(<FullScreenLoadingIndicator {...props} />);
  expect(instance.toJSON()).toMatchSnapshot();
});
