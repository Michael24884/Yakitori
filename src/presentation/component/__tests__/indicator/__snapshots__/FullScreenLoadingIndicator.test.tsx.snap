// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(51,51,51,0.3)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(39,43,50)",
          "borderRadius": 8,
          "height": 80,
          "justifyContent": "center",
          "width": 80,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</Modal>
`;
